src/interpreter.rs

```diff
@@ -18,6 +64,95 @@ impl Execute for Expression {
     }
 }
 
+#[derive(Debug, Clone, Copy)]
+pub enum OperandItem<'a> {
+    Val(bool),
+    Predicate(&'a Predicate),
```

Comment:
I doubt whether it is worthwhile to lazy-eval pred. Pred is quite simple to calculate but copying a pred involves copying strings. Let's examine the actual influence.
IMO lazy-eval works if we go top-down, but it does not make a lot of difference if we always calculate every logic operator.

Reply:
After rethinking, it's only making a reference thus it's acceptable, and the short-circuit always happens at the left-hand side thus it's effectively the same as top-down short-circuit.